xds/internal/xdsclient/xdsresource/resource_type.go

```diff
@@ -58,27 +58,48 @@ type Producer interface {
 // from the xDS server.
 type OnDoneFunc func()
 
-// ResourceWatcher wraps the callbacks to be invoked for different events
-// corresponding to the resource being watched.
+// ResourceWatcher is an interface that can to be implemented to wrap the
+// callbacks to be invoked for different events corresponding to the resource
+// being watched.
```

Comment:
I don't feel like this change to the comment adds much value. The fact that it is an interface is clear from the type definition and since it is an interface, it *needs *to be implemented. So, I would leave this unchanged or improve it in a different way.